tfhe/src/integer/server_key/radix_parallel/neg.rs

```diff
@@ -78,16 +78,8 @@ impl ServerKey {
         &self,
         ctxt: &RadixCiphertext<PBSOrder>,
     ) -> RadixCiphertext<PBSOrder> {
-        let mut tmp_ctxt: RadixCiphertext<PBSOrder>;
-
-        let mut ctxt = if ctxt.block_carries_are_empty() {
-            self.unchecked_neg(ctxt)
-        } else {
-            tmp_ctxt = ctxt.clone();
-            self.full_propagate_parallelized(&mut tmp_ctxt);
-            self.unchecked_neg(&tmp_ctxt)
-        };
-        self.full_propagate_parallelized(&mut ctxt);
-        ctxt
+        let mut flipped = self.bitnot_parallelized(ctxt);
+        self.scalar_add_assign_parallelized(&mut flipped, 1);
+        flipped
```

Comment:
two default calls, could there be a more efficient combination of ops ?

Reply:
Its not that inefficient, since bitnot is just the tipical potentiel pre-emptive carry propagation as all defaults ops do, followed by parallel pbs on each block.

However we can indeed try to do

```
 let mut tmp_ctxt: RadixCiphertext<PBSOrder>;

        let ct = if !ctxt.block_carries_are_empty() {
            tmp_ctxt = ctxt.clone();
            self.full_propagate_parallelized(&mut tmp_ctxt);
            &tmp_ctxt
        } else {
            ctxt
        };

        if self.is_eligible_for_parallel_carryless_add() {
            let mut ct = self.unchecked_neg(ct);
            let generates_or_propagates = self.generate_init_carry_array(&ct);
            let input_carries =
                self.compute_carry_propagation_parallelized_low_latency(generates_or_propagates);
            ct.blocks
                .par_iter_mut()
                .zip(input_carries.par_iter())
                .for_each(|(block, carry_in)| {
                    self.key.unchecked_add_assign(block, carry_in);
                    self.key.message_extract_assign(block);
                });

            ct
        } else {
            let mut ct = self.unchecked_neg(ct);
            self.full_propagate_parallelized(&mut ct);
            ct
        }
```

which basicaly replaces the parallel bps via sequential lwe_additions

Tests seems ok, I'll run benches to see the difference

Reply:
Benches on m6i seems to give:

- 400ms for default_bitnot + default_add(1)
- 350ms for unchecked_neg + parallel carry propagation